clang/lib/Driver/Compilation.cpp

```diff
@@ -128,17 +128,29 @@ bool Compilation::CleanupFile(const char *File, bool IssueErrors) const {
   // able to remove), or non-regular files. Underlying tools may have
   // intentionally not overwritten them.
 
-  // Save the device code files(spv files) only if -fsycl-dump-device-code
-  // option is enabled.
+  // Save the device code files if -fsycl-dump-device-code option is enabled.
   if (TheDriver.isDumpDeviceCodeEnabled()) {
+    bool IsNVPTX = false;
     Arg *DumpDeviceCodeArg =
         getArgs().getLastArg(options::OPT_fsycl_dump_device_code_EQ);
+    // Check if SYCL code is being offloaded to CUDA-enabled GPUs.
+    if (const Arg *SYCLTargets =
+            getArgs().getLastArg(options::OPT_fsycl_targets_EQ)) {
```

Comment:
Is there any way we can support something like `-fsycl-targets=nvptx,spir64` combined?  it would seem that `IsNPVTX` would be set 'true' in this case and will impact the `spir64` values as well.